### ISSUE TITLE
display np.int64 numbers as integers, do not apply 'floatfmt'

### ISSUE THREAD

**Issue by jonsel087** [astanin/python-tabulate#18]
This is more a result of how pandas data frames are stored internally, but the issue comes up when you tabulate any dataframe that contains float type columns. When the df is converted to a numpy array, numpy will upcast *all* int values to floats.

```py
from tabulate import tabulate
import pandas as pd
df = pd.DataFrame({'A':[1,2,3,4],'B':[.20942, .092384, .019823, .01923]})
print(tabulate(df, floatfmt='.2f', headers=df.columns, tablefmt='pipe'))
```
```
|    |    A |    B |
|---:|-----:|-----:|
|  0 | 1.00 | 0.21 |
|  1 | 2.00 | 0.09 |
|  2 | 3.00 | 0.02 |
|  3 | 4.00 | 0.02 |
```

This can further be shown that it's caused by internal storage of `np.int64` dtype:
```py
print(tabulate([['int','float'],[np.int64(1), 3.14159]], headers='firstrow',tablefmt='pipe', floatfmt='.2f'))
```
```
|   int |   float |
|------:|--------:|
|  1.00 |    3.14 |
```

Is there a safe way to convert int64 types to vanilla int when parsing dataframes? Maybe there is a workaround i'm not thinking of.

PS - Note that `np.int64(1) == int(1)` evaluates to `True`

**Comment by MoritzMaxeiner** [comment]
Hey @astanin, I appreciate the package, but just wanted to let you know that pandas DataFrames np.int64 numbers are still evaluated as float when there are other float columns. This is because [this](https://github.com/astanin/python-tabulate/blob/95ae5eb61ef969749e904c90ab429003238d6212/tabulate/__init__.py#L1381) call to `values` causes numpy to upcast all integer columns to float columns. Ideally instead of just using the `values`, each cell would be considered individually.

I'm using a quick and dirty hack to get around this by reapplying the appropriate dtype to each cell afterwards [here](https://github.com/astanin/python-tabulate/blob/95ae5eb61ef969749e904c90ab429003238d6212/tabulate/__init__.py#L1384):

```
rows = [list([tabular_data.dtypes[tabular_data.keys()[i]].type(cell) for i, cell in enumerate(row)]) for row in vals]
```

This is probably unsafe in ways I don't know, but it does allow me to print the issue's primary example correctly (without this hack it still prints the output the original reported posted):

```
from tabulate import tabulate
import pandas as pd
df = pd.DataFrame({'A':[1,2,3,4],'B':[.20942, .092384, .019823, .01923]})
print(tabulate(df, floatfmt='.2f', headers=df.columns, tablefmt='pipe'))
```

```
|    |   A |    B |
|---:|----:|-----:|
|  0 |   1 | 0.21 |
|  1 |   2 | 0.09 |
|  2 |   3 | 0.02 |
|  3 |   4 | 0.02 |
```

**Comment by RobinLangeHyperplan** [comment]
This shouldn't have been closed, the original issue still exists as of 0.10.0:
(when printing dataframes and not arrays)

```
>>> df = pd.DataFrame({'A':[1,2,3,4],'B':[.20942, .092384, .019823, .01923]})
>>> df.dtypes
A      int64
B    float64
dtype: object
>>> print(tabulate(df, floatfmt='.2f', headers=df.columns, tablefmt='pipe'))
|    |    A |    B |
|---:|-----:|-----:|
|  0 | 1.00 | 0.21 |
|  1 | 2.00 | 0.09 |
|  2 | 3.00 | 0.02 |
|  3 | 4.00 | 0.02 |
```
